src/EcoCode.Core/Analyzers/EC91.UseWhereBeforeOrderBy.Fixer .cs

```diff
@@ -0,0 +1,166 @@
+﻿using Microsoft.CodeAnalysis.Formatting;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace EcoCode.Analyzers;
+
+/// <summary>EC91 fixer: Where before Order By.</summary>
+[ExportCodeFixProvider(LanguageNames.CSharp, Name = nameof(UseWhereBeforeOrderByFixer)), Shared]
+public sealed class UseWhereBeforeOrderByFixer : CodeFixProvider
+{
+    /// <inheritdoc/>
+    public override ImmutableArray<string> FixableDiagnosticIds => _fixableDiagnosticIds;
+    private static readonly ImmutableArray<string> _fixableDiagnosticIds = [UseWhereBeforeOrderBy.Descriptor.Id];
+
+    /// <inheritdoc/>
+    public override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+    /// <inheritdoc/>
+     public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+    {
+        var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+        if (root is null) return;
+
+        foreach (var diagnostic in context.Diagnostics)
+        {
+            var parent = root.FindToken(diagnostic.Location.SourceSpan.Start).Parent;
+            if (parent is null) continue;
+            if (diagnostic is null) continue;
+
+            foreach (var node in parent.AncestorsAndSelf())
+            {
+                if (node is QueryExpressionSyntax queryExpression)
+                {
+                    context.RegisterCodeFix(
+                        CodeAction.Create(
+                            title: "With LINQ move 'where' before 'orderby'",
+                            createChangedDocument: c => MoveWhereBeforeOrderBy(context.Document, queryExpression, c),
+                            equivalenceKey: nameof(MoveWhereBeforeOrderBy)),
+                        diagnostic);
+                }
+
+                if (node is InvocationExpressionSyntax invocationExpr)
+                {
+                    context.RegisterCodeFix(
+                        CodeAction.Create(
+                            title: "With LINQ move 'Where' before 'OrderBy'",
+                            createChangedDocument: c => MoveWhereBeforeOrderByWithObject(context.Document, invocationExpr, c),
+                            equivalenceKey: nameof(MoveWhereBeforeOrderByWithObject)),
+                        diagnostic);
+                }
+            }            
+        }
+    }
+
+    private async Task<Document> MoveWhereBeforeOrderBy(Document document, QueryExpressionSyntax queryExpression, CancellationToken cancellationToken)
+    {
+        var whereClause = queryExpression.Body.Clauses.OfType<WhereClauseSyntax>().FirstOrDefault();
```

Comment:
If (queryExpression.Body.Clauses.OfType<WhereClauseSyntax>().FirstOrDefault() is not { } whereClause ||
    queryExpression.Body.Clauses.OfType<OrderByClauseSyntax>().FirstOrDefault() is not { } orderByClause)
{
    return document;
}

Ca simplifie la fonction